src/libsync/owncloudpropagator.cpp

```diff
@@ -388,6 +395,9 @@ void OwncloudPropagator::start(const SyncFileItemVector& items)
                 currentDirJob->append(dir);
             }
             directories.push(qMakePair(item->destination() + "/" , dir));
+        } else if (enableScheduledRequests
+              && (item->_instruction == CSYNC_INSTRUCTION_NEW || item->_instruction == CSYNC_INSTRUCTION_SYNC)) {
+            filesJob->append(item);
```

Comment:
@mrow4a @jturcotte @ogoffart If I'm not mistaken the following is a situation in which that dependency structure could lead to an abort-related problem:

```
Initial state in db and server:
/  - "etag/1"
/A - "etagA1"
/A/F - "etagF1"
```
now when someone touches /A/F on the server, the server state becomes
```
Server state, after touching /A/F
/  - "etag/2"
/A - "etagA2"
/A/F - "etagF2"
```
but if I understand correctly the propagation dependency graph is
```
PropagateDirectory (/)
 |- PropagateDirectory (/A)
 |- PropagateFiles
    |- PropagateItem (/A/F)
```
meaning that /A and /A/F run independently of each other. So it would be possible to completely finish propagating /A before the file transfer /A/F is done. Then aborting the sync run could lead to the db tree
```
/  - "etag/1"
/A - "etagA2"
/A/F - "etagF1"
```
Which means a follow up sync would probably not pick up on /A/F being out of date.

Maybe this hints at a second dependency problem: Currently local and remote MkDir are FullParallelism - so isn't there a chance that with this change one could run into a case where the client wants to download or upload into a non-existant directory?

Possibly I'm missing something, feel free to point out incorrectness.